### DISCUSSION TITLE
Jenkins CI Stages

### DISCUSSION THREAD

**Discussion by mpranj** [ElektraInitiative/libelektra#3926] [Ideas]
Should we add an additional stage with formatting checks etc. before anything else?

It has led to confusion when parallel builds are aborted, as error messages from aborted builds are meaningless. (recently in #3657)

**Comment by markus2330** [comment]
Yes, this is a good idea!

> It has led to confusion when parallel builds are aborted

Probably we should make the first few jobs (release notes, formatting checks) sequential to make the common problems more clear.

**Reply by kodebach** [reply]
I think we could just combine all the basic checks that don't need to build the code (release notes, formatting, typos, man pages, etc.) into one job. The job should always run all the checks and report all results and not abort on the first error.

From my observations of builds, I think this would improve response time when there are many active PRs. AFAICT there is some limit on the number of active jobs, so if there are separate jobs for release notes, formatting, etc. a few PRs could starve out the rest and delay basic feedback.

**Comment by markus2330** [comment]
Can you make an issue from this? The release notes and formatting checks could actually be done without any compilation and should be fast...
